src/components/Calendar/Day.jsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import {PURE} from 'dx-util/src/react/react';
+import moment from 'moment';
+import {CALENDAR_THEME} from './Calendar.constants';
+import noop from '../../util/func/noop';
+import classnames from 'classnames';
+
+@PURE
+export default class Day extends React.Component {
+	static propTypes = {
+		value: React.PropTypes.instanceOf(moment).isRequired,
+		onChange: React.PropTypes.func,
+		dayFormat: React.PropTypes.string.isRequired,
+		isDisabled: React.PropTypes.bool,
+		isCurrent: React.PropTypes.bool,
+		isSelected: React.PropTypes.bool,
+		theme: React.PropTypes.shape(CALENDAR_THEME)
+	}
+
+	static defaultProps = {
+		onChange: noop,
+		isDisabled: false,
+		isCurrent: false,
+		isSelected: false
+	}
+
+	render() {
+		const {
+			theme,
+			value,
+			dayFormat,
+			isCurrent,
+			isDisabled,
+			isSelected
+		} = this.props;
+
+		const className = classnames(theme.day, {
+			[theme.day_disabled]: isDisabled,
+			[theme.day_current]: isCurrent && !isDisabled,
+			[theme.day_selected]: isSelected && !isDisabled
+		});
+
+		return (
+			<div className={className}
+				 onClick={this.onDayClick}>
+				{value.format(dayFormat)}
+			</div>
+		);
+	}
+
+	onDayClick = e => {
+		if (!this.props.isDisabled) {
```

Comment:
if you need to block onclick in disabled state then this should better be a button